pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -193,3 +211,89 @@ func (h *calicoIPPoolHandler) deleteIPPool(endpoint *submV1.Endpoint) error {
 func getEndpointSubnetIPPoolName(endpoint *submV1.Endpoint, subnet string) string {
 	return fmt.Sprintf("submariner-%s-%s", endpoint.Spec.ClusterID, strings.ReplaceAll(subnet, "/", "-"))
 }
+
+func (h *calicoIPPoolHandler) platformIsROKS() (bool, error) {
+	// Submariner GW is deployed on ROKS using LB service with specific annotations.
+	service, err := h.k8sClient.CoreV1().Services(h.namespace).Get(context.TODO(), gwLBSvcName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return false, nil
+	}
+
+	if err != nil {
+		return false, errors.Wrap(err, "error reading gw lb service")
+	}
+
+	return service.GetAnnotations()[gwLBSvcROKSAnnotation] != "", nil
+}
+
+// workaround to address datapath issue with default Calico IPPool configuration for ROKS platform,
+// IPIPMode of calico default IPPool should be set to 'Always'.
+func (h *calicoIPPoolHandler) updateROKSCalicoCfg() error {
+	isROKS, err := h.platformIsROKS()
+	if err != nil {
+		return err
+	}
+
+	if !isROKS {
+		return nil
+	}
+
+	// platform is ROKS, make sure that IPIPMode of default IPPool is Always
+
+	err = util.Update(context.TODO(), h.iPPoolResourceInterface(), &calicoapi.IPPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: defaultV4IPPoolName,
+		},
+	}, func(existing *calicoapi.IPPool) (*calicoapi.IPPool, error) {
+		if existing.Spec.IPIPMode == calicoapi.IPIPModeAlways {
+			logger.Infof("IPIPMode of %s IPPool already set to Always", defaultV4IPPoolName)
+			return existing, nil // no need to update
+		}
+
+		if existing.Labels == nil {
+			existing.Labels = make(map[string]string)
+		}
+
+		if existing.Annotations == nil {
+			existing.Annotations = make(map[string]string)
+		}
+
+		// mark 'submariner-managed' label for persistency
+		existing.Labels[submarinerManagedLabel] = "true"
+		existing.Annotations[submarinerPrevIPIPMode] = string(existing.Spec.IPIPMode)
+		existing.Spec.IPIPMode = calicoapi.IPIPModeAlways
+		logger.Infof("IPIPMode of %s IPPool successfully set to Always", defaultV4IPPoolName)
```

Comment:
Technically it hasn't been successfully set yet 😄 

```suggestion
		logger.Infof("Setting IPIPMode of %s IPPool to Always", defaultV4IPPoolName)
```

Reply:
Thanks